src/build/__init__.py

```diff
@@ -1,387 +1,31 @@
-        """
-        # prepare_metadata hook
-        metadata = self.prepare('wheel', output_directory)
-        if metadata is not None:
-            return metadata
-
-        # fallback to build_wheel hook
-        wheel = self.build('wheel', output_directory)
-        match = parse_wheel_filename(os.path.basename(wheel))
-        if not match:
-            msg = 'Invalid wheel'
-            raise ValueError(msg)
-        distinfo = f"{match['distribution']}-{match['version']}.dist-info"
-        member_prefix = f'{distinfo}/'
-        with zipfile.ZipFile(wheel) as w:
-            w.extractall(
-                output_directory,
-                (member for member in w.namelist() if member.startswith(member_prefix)),
-            )
-        return os.path.join(output_directory, distinfo)
-
-    def _call_backend(
-        self, hook_name: str, outdir: PathType, config_settings: ConfigSettingsType | None = None, **kwargs: Any
-    ) -> str:
-        outdir = os.path.abspath(outdir)
-
-        callback = getattr(self._hook, hook_name)
-
-        if os.path.exists(outdir):
-            if not os.path.isdir(outdir):
-                msg = f"Build path '{outdir}' exists and is not a directory"
-                raise BuildException(msg)
-        else:
-            os.makedirs(outdir)
-
-        with self._handle_backend(hook_name):
-            basename: str = callback(outdir, config_settings, **kwargs)
-
-        return os.path.join(outdir, basename)
-
-    @contextlib.contextmanager
-    def _handle_backend(self, hook: str) -> Iterator[None]:
-        try:
-            yield
-        except pyproject_hooks.BackendUnavailable as exception:
-            raise BuildBackendException(
-                exception,
-                f"Backend '{self._backend}' is not available.",
-                sys.exc_info(),
-            ) from None
-        except subprocess.CalledProcessError as exception:
-            raise BuildBackendException(exception, f'Backend subprocess exited when trying to invoke {hook}') from None
-        except Exception as exception:
-            raise BuildBackendException(exception, exc_info=sys.exc_info()) from None
-
-    @staticmethod
-    def log(message: str) -> None:
-        """
-        Log a message.
-
-        The default implementation uses the logging module but this function can be
-        overridden by users to have a different implementation.
-
-        :param message: Message to output
-        """
-        _logger.log(logging.INFO, message, stacklevel=2)
 
+__version__ = '1.1.1'
 
 __all__ = [
     '__version__',
-    'BuildSystemTableValidationError',
     'BuildBackendException',
     'BuildException',
+    'BuildSystemTableValidationError',
+    'check_dependency',
     'ConfigSettingsType',
+    'DistributionType',
     'FailedProcessError',
     'ProjectBuilder',
     'RunnerType',
     'TypoWarning',
-    'check_dependency',
 ]
-
-
-def __dir__() -> list[str]:
-    return __all__
```

Comment:
We shouldn't lose this, it keeps things like `annotations` from showing up in tab completion, and is needed for mypy re-exports.

Reply:
Needed for mypy re-exports - what do you mean?

I removed it because the `__init__` module doesn't contain anything other than what's exported, barring `annotations`.

Reply:
(Restored.)

Reply:
I’m not fussed either way - I’ve put it back 😄